### PR TITLE
feat: JSR package

### PR DESCRIPTION
**Pull request by nzakas**
Creates a JSR package for `@eslint/markdown`:

- Added `jsr.json`
- Added JSR publish to `release-please.yml`
- Added JSR test to `ci.yml`

Refs #260

### CODE REVIEW

**Comment by nzakas** [comment]
Ping @eslint/eslint-team

File: jsr.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@eslint/markdown",
+  "version": "6.0.0",
+  "exports": "./dist/esm/index.js",
+  "publish": {
+    "include": [
+      "dist/esm/index.js",
+      "dist/esm/index.d.ts",
+      "dist/esm/types.ts",
```
Comment (mdjermanovic) [inline jsr.json:9]:
```suggestion
      "dist/esm/types.ts",
      "dist/esm/types.d.ts",
```

Should maybe `dist/esm/types.d.ts` also be published?

**Comment by mdjermanovic** [comment]
There are merge conflicts now, caused by the prettier commit.

File: jsr.json (hunk 1 above)
Comment (nzakas) [reply jsr.json:9]: I'm not sure, but doesn't hurt to include it.

**Comment by mdjermanovic** [comment]
Still merge conflicts.

File: jsr.json (hunk 1 above)
Comment (mdjermanovic) [inline jsr.json:9]:
```suggestion
      "dist/esm/types.ts",
      "dist/esm/types.d.ts",
```

Seems that https://github.com/eslint/markdown/commit/c29d88da600f0a7b35d1514552f9cc107da6b8a3 was lost in rebase.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**